libraries/ms-common/inc/ms_test_helpers.h

```diff
@@ -27,6 +28,7 @@
     MS_TEST_HELPER_AWAIT_EVENT(e);                             \
     TEST_ASSERT_EQUAL((tx_event).id, e.id);                    \
     TEST_ASSERT_TRUE(fsm_process_event(CAN_FSM, &(tx_event))); \
+    delay_us(750);                                             \
```

Comment:
Necessary as previously;

TX -> RX ->  Raise `LOW` Priority Event In Handler -> TX -> Read `LOW` Priority Event from handler` -> RX was occurring.

This allows the RX to flush before any events raised in the handler are processed and prevents races in the case of events of different priorities. Note that in the case of all `NORMAL` priority events this will actually be TX -> RX -> EVENT -> TX -> RX.

Reply:
👍 

I noticed that your helper passes `[tx|rx]_event` into `fsm_process_event` instead of `e`. Is there a reason for that? Since you're only comparing the event IDs, you could do something like

```
TEST_ASSERT_EQUAL((tx_event), e.id);
TEST_ASSERT_TRUE(fsm_process_event(CAN_FSM, &e));
```

instead. That would remove the need for your static TX/RX events.